smart_card_connector_app/src/chrome-api-provider.js

```diff
@@ -265,6 +265,51 @@ function convertDispositionToNumber(disposition) {
   }
 }
 
+/**
+ * Converts a chrome.smartCardProviderPrivate.Protocol to a predefined
+ * Protocol Control Information structure used in PCSC API.
+ * @param {!chrome.smartCardProviderPrivate.Protocol} protocol
+ * @returns {!PcscApi.SCARD_IO_REQUEST}
+ * @throws {Error} Throws error if unknown protocol is encountered.
+ */
+function convertProtocolToPci(protocol) {
+  switch (protocol) {
+    case chrome.smartCardProviderPrivate.Protocol.T0:
+      return PcscApi.SCARD_PCI_T0;
+    case chrome.smartCardProviderPrivate.Protocol.T1:
+      return PcscApi.SCARD_PCI_T1;
+    case chrome.smartCardProviderPrivate.Protocol.RAW:
+      return PcscApi.SCARD_PCI_RAW;
+    default:
+      throw new Error(`No predefined PCI structure for protocol ${protocol}`);
+  }
+}
+
+/**
+ * Converts from a PCSC API value to a
+ * chrome.smartCardProviderPrivate.ConnectionState. We only take the highest bit
+ * of the value, since it contains all the needed information, e.g.
+ * SCARD_NEGOTIABLE implies that SCARD_POWERED and SCARD_PRESENT are also set.
```

Comment:
I double-checked PC/SC-Lite sources and it seems to be true. However, as extra reassurance, maybe log a warning in case we observe a combination we didn't expect in this function?